core/server/api_container/server/startosis_engine/kurtosis_starlark_framework/enclave_plan_capabilities/enclave_plan_capabilites_builder.go

```diff
@@ -0,0 +1,59 @@
+package enclave_plan_capabilities
+
+import (
+	"github.com/kurtosis-tech/kurtosis/container-engine-lib/lib/backend_interface/objects/service"
+)
+
+type EnclavePlanCapabilitiesBuilder struct {
+	instructionTypeStr string
+	serviceName        service.ServiceName
+	serviceNames       []service.ServiceName
+	artifactName       string
+	filesArtifactMD5   []byte
```

Comment:
I think it would be less confusing to keep the mapping file artifact -> file MD5. Here we're losing the relationship info, which right now it surprisingly not a blocker, but might become on. And generally speaking I prefer not creating too much entropy